cli/flox-rust-sdk/src/models/manifest/typed.rs

```diff
@@ -456,7 +456,18 @@ pub struct ManifestPackageDescriptorStorePath {
     pub(crate) priority: Option<u64>,
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize, Default, PartialEq, Eq, Hash)]
+#[derive(
+    Debug,
+    Clone,
+    Serialize,
+    Deserialize,
+    Default,
+    PartialEq,
+    Eq,
+    Hash,
+    derive_more::Deref,
```

Comment:
I'm not a _huge_ fan of using `Deref` for type conversions or accessing inner variables like this. This is also listed as an anti-pattern in the (unofficial) design patterns "book": https://rust-unofficial.github.io/patterns/anti_patterns/deref.html#discussion.

I'd be fine with `inner` and `inner_mut` methods that return the map. I think that's slightly better than `foo.0` because it doesn't rely on the structure of `foo` itself. If we change `foo` to have a named `map` field, the `foo.0` call sites break, whereas the `inner(_mut)` callsites won't.

Reply:
Done (will squash)